### ISSUE TITLE
Store port version as a property

### ISSUE THREAD

**Issue by mojca** [macports-gsoc/macports-buildbot#5]
It would be useful to store port version as a separate property.

I suspect that something like the following could do the trick:
```python
addStep(steps.SetPropertyFromCommand(
    command=['port', 'info', '--version', '--line', portname],
    property='version'))
```

It would in fact be nice to also add this to the existing 0.8 buildbot setup to make the information for the app more useful.

@arjunsalyan

**Comment by rajgoesout** [comment]
Just to be clear, for [this](https://github.com/macports/macports-ports/blob/master/editors/abiword/Portfile) port the version is 2.4.5. Does it support any other version?
I mean, when a port gets updated to a new version, can a user install its old version?

For something like python ports (like https://github.com/macports/macports-ports/blob/master/python/py-tensorflow/Portfile) does 'version' refer to python version (27,36,37) or port [version](https://github.com/macports/macports-ports/blob/master/python/py-tensorflow/Portfile#L14).

**Comment by mojca** [comment]
Yes, the port version is 2.4.5, no other version is relevant in any way.
When a port gets updated to a newer version, the only way to install an old version is:
* to either activate a version that has been installed before on that same computer (and only deactivated rather than completely removed)
* or modify the Portfile, either manually for one port, or by checking out an older git commit.

Python version is not relevant for ports like `py-tensorflow`. We have subports `py27-tensorflow`, `py36-tensorflow`, `py37-tensorflow`. All of them have version `1.14.0_2` (the `_2` part is revision which I would also include). The same is not true for py-django though:
```
> port info --version --line py27-django
1.11.21
> port info --version --line py37-django
2.2.2
```
but those are two different ports.

I would say that you don't need to worry about any of those details. Just running
```
 port info --version --line py27-tensorflow
```
or maybe
```
 port info --version --revision --line py27-tensorflow
```
just before running the actual build should likely be all that really matters.

**Comment by rajgoesout** [comment]
> I suspect that something like the following could do the trick:

I guess this wouldn't do the trick , because portname is assigned when the build is in progress, not before.

**Comment by mojca** [comment]
It gets assigned when you trigger the build. How else would you run `port install whatever` inside the build? You would call this before running "random" script in port build and it would assign property to current job.

**Comment by rajgoesout** [comment]
https://github.com/macports-gsoc/macports-buildbot/blob/master/master.cfg#L142

**Comment by mojca** [comment]
No, around line 188.

**Comment by rajgoesout** [comment]
Thanks 🙂
